devito/passes/iet/languages/C.py

```diff
@@ -55,3 +56,13 @@ class CPrinter(BasePrinter, C99CodePrinter):
 
     def _print_ImaginaryUnit(self, expr):
         return '_Complex_I'
+
+
+class PetscCPrinter(CPrinter):
+    _restrict_keyword = ''
+
+    # TODO: Check to see whether Petsc is compiled with
+    # 32-bit or 64-bit integers
```

Comment:
Also need to check whether `PetscScalar` is `float` or `double` here. Float and double can't both map to PetscScalar